tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -150,59 +159,101 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     };
 
     fluid.tests.tooltipModuleSource = function (pager) {
-        var pageLinksTop = $("a", pager.pagerBar.locate("pageLinks"));
-        var pageLinksBottom = $("a", pager["pagerBar-1"].locate("pageLinks"));
-
-        var tooltipContents = [
-            [
-                {nodeName: "b", nodeText: "bird"},
-                {nodeName: "b", nodeText: "fish"}
+        
+        var expectedTooltipContents = {
+            size2: [
+                [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "cat"} ],
+                [ {nodeName: "b", nodeText: "bird"}, {nodeName: "b", nodeText: "fish"} ],
+                [ {nodeName: "b", nodeText: "camel"}, {nodeName: "b", nodeText: "dragon"} ],
+                [ {nodeName: "b", nodeText: "ant"}, {nodeName: "b", nodeText: "ant"} ]
+            ],
+            size3: [
+                [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "bird"} ],
+                [ {nodeName: "b", nodeText: "fish"}, {nodeName: "b", nodeText: "dragon"} ],
+                [ {nodeName: "b", nodeText: "ant"}, {nodeName: "b", nodeText: "ant"} ]
+            ],
+            size4: [
+                [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "fish"} ],
+                [ {nodeName: "b", nodeText: "camel"}, {nodeName: "b", nodeText: "ant"} ]
             ]
-        ];
+        };
 
         var sequence = [];
 
-        var assertVisibleTips = function (message, targetIds) {
+        var assertVisibleTips = function (message, targetIds, expectedTooltipContents, index) {
             sequence.push({
                 event: "{trackTooltips}.events.notifyFocusChange",
                 listener: function () {
                     fluid.tests.tooltip.assertVisible(message, pager, targetIds, null, function (tooltip) {
-                        jqUnit.assertNode("The contents of the tooltip should be set", tooltipContents[0], $("b", tooltip));
+                        jqUnit.assertNode("When the page size is " + pager.model.pageSize + ", The contents of the tooltip should be set", expectedTooltipContents[index], $("b", tooltip));
                     });
                 }
             });
         };
 
-        var tooltipTest = function (location) {
-            return function (idx, linkEl) {
-                var linkId = linkEl.id;
-                var link = $(linkEl);
+        var tooltipTest = function (location, index, expectedTooltipContents) {
+            var link =  "#rendered-ioc .flc-pager-" + location + " .flc-pager-links .flc-pager-pageLink a:eq(" + index + ")";
+            var linkNumber = index + 1;
+            var linkId = "page-link:link" + linkNumber + (location === "bottom" ? "-1" : "");
+            sequence.push({
```

Comment:
I found it's difficult to follow all these dynamically generated test sequence blocks. It defeats the purpose of having well outlined test sequences with IoC testing framework - http://docs.fluidproject.org/infusion/development/IoCTestingFramework.html

In your case that does not require to hook up event listeners or model listeners on the fly, like what line 184-191 does, as well as not having asynchronous calls, IoC testing framework might not be necessary. Consider to use jqunit.test() directly.

Also, instead of adding to existing fluid.tests.tooltipModuleSource() function, consider using a separate test as you were meant to cover another use case.